src/components/TrendingTrack.js

```diff
@@ -0,0 +1,62 @@
+import React, { useState } from "react";
+import styles from "./TrendingTrack.module.css";
+import playButton from "../images/play.svg";
+import likeButton from "../images/heart.svg";
+import moreButton from "../images/dots.svg";
+
+const TrendingTrack = ({ track }) => {
+  const [hovered, setHovered] = useState(false);
+
+  const handleHover = (e) => {
+    e.type === "mouseenter" ? setHovered(true) : setHovered(false);
+  };
+  return (
+    <>
+      <div className={styles.trackContainer}>
+        <div
+          className={styles.trackItem}
+          onMouseEnter={handleHover}
+          onMouseLeave={handleHover}
+          onClick={() => {
+            console.log("image");
+          }}
+        >
+          <img src={track.cover} />
+          {hovered && (
+            <>
+              <div
+                className={styles.playButton}
+                onClick={(e) => {
+                  e.stopPropagation();
+                  console.log("play");
+                }}
+              >
+                <img src={playButton} />
+              </div>
+              <div className={styles.likeButton}>
+                <img
+                  src={likeButton}
+                  onClick={(e) => {
+                    e.stopPropagation();
+                    console.log("like");
+                  }}
+                />
+                <img
+                  src={moreButton}
+                  onClick={(e) => {
+                    e.stopPropagation();
+                    console.log("more");
+                  }}
+                />
+              </div>
+            </>
+          )}
```

Comment:
If you enclose all this in a div and give that a position of absolute and a width and height of 100%, you can then give it display flex and just layout all the internal elements normally. Let me know if that makes sense - if not I can work up an example